### PR TITLE
[1LP][RFR] - Service Dialog test

### PR DESCRIPTION
**Pull request by dgaikwad**
**Adding new test -** Test to check `Load values on init` option in service dialog when `Dynamix` box enable

### PRT Run

{{pytest: cfme/tests/services/test_dialog_element_in_catalog.py::test_catalog_load_values_on_init  --long-running -v}}

### CODE REVIEW

File: cfme/tests/services/test_dialog_element_in_catalog.py [hunk 1]
```diff
@@ -774,13 +787,30 @@ def test_load_values_on_init_option_service_dialog_element():
...
+
+        }
+    }
+    dialog, element = create_dialog(appliance, element_data)
+    view = appliance.browser.create_view(EditElementView)
+    view.element.edit_element(element_data['element_information']['ele_label'])
+    assert view.element_information.dynamic_chkbox.is_enabled
+    view.options.click()
+    assert view.options.load_values_on_init.is_enabled
+    request.addfinalizer(dialog.delete_if_exists)
```
Comment (ganeshhubale) [inline cfme/tests/services/test_dialog_element_in_catalog.py]: Put ```request.addfinalizer(dialog.delete_if_exists)``` above the assert. If assert gets failed then it will not work.

File: cfme/tests/services/test_dialog_element_in_catalog.py [hunk 2]
```diff
@@ -774,13 +787,30 @@ def test_load_values_on_init_option_service_dialog_element():
...
+            'ele_label': fauxfactory.gen_alphanumeric(15, start="ele_label_"),
+            'ele_name': fauxfactory.gen_alphanumeric(15, start="ele_name_"),
+            'ele_desc': fauxfactory.gen_alphanumeric(15, start="ele_desc_"),
+            'choose_type': "Text Box",
+            'dynamic_chkbox': True
+
+        }
+    }
+    dialog, element = create_dialog(appliance, element_data)
+    view = appliance.browser.create_view(EditElementView)
```
Comment (ganeshhubale) [inline cfme/tests/services/test_dialog_element_in_catalog.py]: can we make ```create_dialog``` - fixture if it is setup.

File: cfme/tests/services/test_dialog_element_in_catalog.py [hunk 3]
```diff
@@ -86,6 +86,19 @@ def custom_categories(appliance):
...
+    if not label:
+        label = fauxfactory.gen_alphanumeric(15, start="label_")
+    sd = service_dialog.create(label=label, description="my dialog")
+    tab = sd.tabs.create(tab_label=fauxfactory.gen_alphanumeric(start="tab_"),
+                         tab_desc="my tab desc")
+    box = tab.boxes.create(box_label=fauxfactory.gen_alphanumeric(start="box_"),
+                           box_desc="my box desc")
+    element = box.elements.create(element_data=[element_data])
+    return sd, element
+
```
Comment (ganeshhubale) [inline cfme/tests/services/test_dialog_element_in_catalog.py]: Instead of this function. please check if you can use this fixture in this file: https://github.com/ManageIQ/integration_tests/pull/9995/files#diff-faeac2d3d489ac01e0ebae6983c9c3cbR36

Comment (dgaikwad) [reply cfme/tests/services/test_dialog_element_in_catalog.py]: Actually I do not want to pass same input to `create_dialog` so kept it as a function instead of fixture

File: cfme/tests/services/test_dialog_element_in_catalog.py [hunk 4]
```diff
@@ -759,10 +772,10 @@ def test_access_child_services_from_the_my_service():
     pass
 
 
-@pytest.mark.meta(coverage=[1684575])
-@pytest.mark.manual
+@pytest.mark.ignore_stream("5.10")
+@pytest.mark.meta(automates=[1684575])
 @pytest.mark.tier(2)
-def test_load_values_on_init_option_service_dialog_element():
```
Comment (ganeshhubale) [inline cfme/tests/services/test_dialog_element_in_catalog.py]: Remove tier marker. It is already defined globally: https://github.com/ManageIQ/integration_tests/pull/9995/files#diff-faeac2d3d489ac01e0ebae6983c9c3cbR21

File: cfme/tests/services/test_dialog_element_in_catalog.py [hunk 5]
```diff
@@ -774,13 +787,30 @@ def test_load_values_on_init_option_service_dialog_element():
...
+
+    element_data = {
+        'element_information': {
+            'ele_label': fauxfactory.gen_alphanumeric(15, start="ele_label_"),
+            'ele_name': fauxfactory.gen_alphanumeric(15, start="ele_name_"),
+            'ele_desc': fauxfactory.gen_alphanumeric(15, start="ele_desc_"),
+            'choose_type': "Text Box",
+            'dynamic_chkbox': True
+
+        }
```
Comment (ganeshhubale) [inline cfme/tests/services/test_dialog_element_in_catalog.py]: remove blank line(804, 796)

File: cfme/tests/services/test_dialog_element_in_catalog.py (hunk 3 above)
Comment (ganeshhubale) [reply cfme/tests/services/test_dialog_element_in_catalog.py]: - you are already passing input ```dialog, element = create_dialog(appliance, element_data)```. 
- Also, for now, I can see this function used for this test case only, Want to know use of **label=None** argument here? or we can directly provide ```sd = service_dialog.create(label=fauxfactory.gen_alphanumeric(15, start="label_"), description="my dialog")```

Comment (dgaikwad) [reply cfme/tests/services/test_dialog_element_in_catalog.py]: I have updated the function signature now

**Comment by john-dupuy** [comment]
@ganeshhubale can we move this to 1LP?

File: cfme/tests/services/test_dialog_element_in_catalog.py (hunk 4 above)
Comment (ganeshhubale) [reply cfme/tests/services/test_dialog_element_in_catalog.py]: @dgaikwad any update on this comment?

**Comment by dgaikwad** [comment]
@ganeshhubale required changes are done, please review PR.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

File: cfme/tests/services/test_dialog_element_in_catalog.py [hunk 6]
```diff
@@ -88,6 +87,18 @@ def custom_categories(appliance):
     category.delete_if_exists()
 
 
+def create_dialog(appliance, element_data):
```
Comment (digitronik) [inline cfme/tests/services/test_dialog_element_in_catalog.py]: is it possible to use fallowing fixture?
https://github.com/ManageIQ/integration_tests/pull/9995/files#diff-faeac2d3d489ac01e0ebae6983c9c3cbL38

File: cfme/tests/services/test_dialog_element_in_catalog.py [hunk 7]
```diff
@@ -776,13 +786,28 @@ def test_load_values_on_init_option_service_dialog_element():
...
+            'choose_type': "Text Box",
+            'dynamic_chkbox': True
+        }
+    }
+    dialog, element = create_dialog(appliance, element_data)
+    view = appliance.browser.create_view(EditElementView)
+    view.element.edit_element(element_data['element_information']['ele_label'])
+    assert view.element_information.dynamic_chkbox.is_enabled
+    view.options.click()
+    request.addfinalizer(dialog.delete_if_exists)
```
Comment (digitronik) [inline cfme/tests/services/test_dialog_element_in_catalog.py]: Best practice: Always put finalizer after step of creation. here I think your dialog creating at L804.

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 66422ab397cc55546ff4107d9f531ac6f129f44c

The *local* fixture **`service_dialog`** is used in the following files:
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *catalog_item_local*
    - *test_required_dialog_elements*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 9104c441479ab3d4abf3cdc5d07d11ead7273e73

The *local* fixture **`service_dialog`** is used in the following files:
- **cfme/tests/services/test_dialog_element_in_catalog.py**
    - *catalog_item_local*
    - *test_required_dialog_elements*
    - *test_catalog_load_values_on_init*

Please, consider creating a PRT run to make sure your fixture changes do not break existing usage :smiley:

File: cfme/tests/services/test_dialog_element_in_catalog.py (hunk 6 above)
Comment (dgaikwad) [reply cfme/tests/services/test_dialog_element_in_catalog.py]: Hello @digitronik , I have tried with fixture which you mentioned here but this fixture is test case-specific fixture so can not use as normal generic fixture. So now instead of using normal function I have created one generic fixture and automated testcase. from now new testcase also can use new implemented `create_service_dialog` fixture.

File: cfme/tests/services/test_dialog_element_in_catalog.py [hunk 8]
```diff
@@ -718,13 +746,18 @@ def test_load_values_on_init_option_service_dialog_element():
         casecomponent: Services
         initialEstimate: 1/16h
         testSteps:
-            1. create a service dialog
+            1. create a service dialog with option dynamic to True
```
Comment (digitronik) [inline cfme/tests/services/test_dialog_element_in_catalog.py]: It looks like part of test steps not setup. If yes, I would try to create dynamic service dialog as part of test only.
Second thought if it setup; we already have one fixture `service_dialog` better try to modify that in state of creating new fixture.

Comment (dgaikwad) [reply cfme/tests/services/test_dialog_element_in_catalog.py]: dialog creation is part of testcase so updated tc accordingly.
